openmc/deplete/microxs.py

```diff
@@ -0,0 +1,235 @@
+            else:
+                xs[rxn] = ArbitraryXS(rxn, domain=reaction_domain, groups=groups, by_nuclide=True)
+            tallies += xs[rxn].tallies.values()
+
+        model.tallies = tallies
+
+        # create temporary run
+        with tempfile.TemporaryDirectory() as temp_dir:
+            statepoint_path = model.run(cwd=temp_dir)
+
+            with StatePoint(statepoint_path) as sp:
+                for rxn in xs:
+                    xs[rxn].load_from_statepoint(sp)
+
+        # Build the DataFrame
+        micro_xs = cls()
+        for rxn in xs:
+            df = xs[rxn].get_pandas_dataframe(xs_type='micro')
+            df.index = df['nuclide']
+            df.drop(['nuclide', xs[rxn].domain_type, 'group in', 'std. dev.'], axis=1, inplace=True)
+            df.rename({'mean': rxn}, axis=1, inplace=True)
+            micro_xs = concat([micro_xs, df], axis=1)
+
+        # Revert to the original tallies and materials
+        model.tallies = original_tallies
+        model.materials = original_materials
+
+        return micro_xs
+
+    @classmethod
+    def _add_dilute_nuclides(cls, chain_file, model, dilute_initial):
+        """
+        Add nuclides not present in burnable materials that have neutron data
+        and are present in the depletion chain to those materials. This allows
+        us to tally those specific nuclides for reactions to create one-group
+        cross sections.
+
+        Parameters
+        ----------
+        chain_file : str
+            Path to the depletion chain XML file that will be used in depletion
+            simulation. Used to determine cross sections for materials not
+            present in the inital composition.
+        model : openmc.Model
+            Model object
+        dilute_initial : float
+            Initial atom density [atoms/cm^3] to add for nuclides that
+            are zero in initial condition to ensure they exist in the cross
+            section data. Only done for nuclides with reaction rates.
+
+        Returns
+        -------
+        reactions : list of str
+            List of reaction names
+        diluted_materials : openmc.Materials
+            :class:`openmc.Materials` object with nuclides added to burnable
+            materials.
+        """
+        chain = Chain.from_xml(chain_file)
+        reactions = chain.reactions
+        cross_sections = _find_cross_sections(model)
+        nuclides_with_data = _get_nuclides_with_data(cross_sections)
+        burnable_nucs = [nuc.name for nuc in chain.nuclides
+                         if nuc.name in nuclides_with_data]
+        diluted_materials = Materials()
+        reset_auto_ids()
+        for material in model.materials:
+            if material.depletable:
+                nuc_densities = material.get_nuclide_atom_densities()
+                dilute_density = 1.e-24 * dilute_initial
+                diluted_material = Material()
```

Comment:
Rather than use `reset_auto_ids` (which itself is another undesired side effect), let's just silence the warning from creating another material with the same ID:
```Python
import warnings

with warnings.catch_warnings():
    warnings.simplefilter("ignore", openmc.IDWarning)
    diluted_material = Material(material_id=material.id)
...
```